src/project/view/scorepropertiesmodel.cpp

```diff
@@ -0,0 +1,220 @@
+        m_tags.append({ tagName, additionalTags[tagName].toString(), false });
+    }
+
+    endResetModel();
+}
+
+QVariant ScorePropertiesModel::data(const QModelIndex& index, int role) const
+{
+    if (!index.isValid() || index.row() >= rowCount()) {
+        return QVariant();
+    }
+
+    Tag tag = m_tags[index.row()];
+
+    switch (role) {
+    case TagName:
+        return tag.name;
+    case TagValue:
+        return tag.value;
+    case IsStandardTag:
+        return tag.isStandardTag;
+    }
+
+    return QVariant();
+}
+
+bool ScorePropertiesModel::setData(const QModelIndex& index, const QVariant& value, int role)
+{
+    if (!index.isValid() || index.row() >= rowCount()) {
+        return false;
+    }
+
+    Tag& tag = m_tags[index.row()];
+
+    switch (role) {
+    case TagName:
+        if (!value.canConvert<QString>()) {
+            return false;
+        }
+        tag.name = value.toString();
+        return true;
+    case TagValue:
+        if (!value.canConvert<QString>()) {
+            return false;
+        }
+        tag.value = value.toString();
+        return true;
+    default:
+        return false;
+    }
+}
+
+int ScorePropertiesModel::rowCount(const QModelIndex&) const
+{
+    return m_tags.size();
+}
+
+QHash<int, QByteArray> ScorePropertiesModel::roleNames() const
+{
+    static const QHash<int, QByteArray> roles {
+        { TagName, "tagName" },
+        { TagValue, "tagValue" },
+        { IsStandardTag, "isStandardTag" }
+    };
+
+    return roles;
+}
+
+QString ScorePropertiesModel::filePath() const
+{
+    return m_projectMetaInfo.filePath.toQString();
+}
+
+QString ScorePropertiesModel::version() const
+{
+    return m_projectMetaInfo.musescoreVersion;
+}
+
+QString ScorePropertiesModel::revision() const
+{
+    int rev = m_projectMetaInfo.musescoreRevision;
+    QString revision;
+    if (rev > 99999) {     // MuseScore 1.3 is decimal 5702, 2.0 and later uses a 7-digit hex SHA
+        revision = QString::number(rev, 16);
+    } else {
+        revision = QString::number(rev, 10);
+    }
+
+    return revision;
+}
+
+QString ScorePropertiesModel::apiLevel() const
+{
+    return QString::number(m_projectMetaInfo.mscVersion);
+}
+
+void ScorePropertiesModel::newTag()
+{
+    int destinationIndex = m_tags.size();
+    beginInsertRows(QModelIndex(), destinationIndex, destinationIndex);
+
+    Tag tag = { "Property", "Value", false };
+    m_tags.append(tag);
+
+    endInsertRows();
+
+    emit tagAdded(destinationIndex);
+}
+
+void ScorePropertiesModel::deleteTag(int index)
+{
+    if (index < 0 && index >= m_tags.size()) {
+        return;
+    }
+
+    beginRemoveRows(QModelIndex(), index, index);
+
+    m_tags.removeAt(index);
+
+    endRemoveRows();
+}
+
+void ScorePropertiesModel::saveTags()
+{
+    INotationProjectPtr project = context()->currentProject();
+    if (!project) {
+        return;
+    }
+
+    ProjectMeta meta;
+
+    meta.title = m_tags[0].value;
```

Comment:
Here we have the same problem we had before in qml. You should use a more explicit way to access the tags (instead of using just array indexes)